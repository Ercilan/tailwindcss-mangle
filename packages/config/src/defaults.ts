import process from 'node:process'
import { defaultMangleClassFilter } from '@tailwindcss-mangle/shared'
import { CSS_LANGS_RE } from 'is-css-request'
import type { MangleUserConfig, PatchUserConfig, UserConfig } from './types'

const defaultPipelineInclude = ['**/*.{html,js,ts,jsx,tsx,vue,svelte,astro,elm,php,phtml,mdx,md}', CSS_LANGS_RE]

const defaultPipelineExclude = [/[\\/](node_modules|dist|\.temp|\.cache|\.vscode)[\\/]/]

export function getDefaultPatchConfig(): PatchUserConfig {
  return {
    output: {
      filename: '.tw-patch/tw-class-list.json',
      removeUniversalSelector: true,
      loose: true,
    },
    tailwindcss: {},
  }
}

export function getDefaultMangleUserConfig(): MangleUserConfig {
  return {
    mangleClassFilter: defaultMangleClassFilter,
    include: defaultPipelineInclude,
    exclude: defaultPipelineExclude,
    disabled: process.env.NODE_ENV === 'development',
    classListPath: '.tw-patch/tw-class-list.json',
    classMapOutput: {
      enable: false,
      filename: '.tw-patch/tw-map-list.json',
      loose: true,
    },
    preserveFunction: [],
  }
}

export function getDefaultUserConfig(): UserConfig {
  return {
    patch: getDefaultPatchConfig(),
    mangle: getDefaultMangleUserConfig(),
  }
}
