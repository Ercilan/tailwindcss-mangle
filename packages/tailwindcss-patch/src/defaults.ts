import type { DeepRequired, InternalPatchOptions, PatchOptions } from './types'
import process from 'node:process'
import { defu } from '@tailwindcss-mangle/shared'

export function getDefaultPatchOptions(): DeepRequired<PatchOptions> {
  return {
    applyPatches: {
      exportContext: true,
      extendLengthUnits: false,
    },
    overwrite: true,
  }
}

export function getPatchOptions(options?: PatchOptions) {
  return defu<InternalPatchOptions, Partial<InternalPatchOptions>[]>(
    options,
    {
      basedir: process.cwd(),
    },
    getDefaultPatchOptions(),
  )
}
