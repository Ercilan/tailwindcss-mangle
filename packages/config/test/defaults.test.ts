import { getDefaultPatchConfig, getDefaultUserConfig } from '@/defaults'
import { createFilter } from '@rollup/pluginutils'
import { omit } from 'lodash-es'

function omitCwdPath(o: any) {
  return omit(o, ['tailwindcss.cwd', 'patch.tailwindcss.cwd'])
}

describe('defaults', () => {
  it('getDefaultPatchConfig', () => {
    expect(omitCwdPath(getDefaultPatchConfig())).toMatchSnapshot()
  })

  it('getDefaultUserConfig', () => {
    expect(omitCwdPath(getDefaultUserConfig())).toMatchSnapshot()
  })
})

describe('createFilter', () => {
  it('case 0', () => {
    const config = getDefaultUserConfig()
    const filter = createFilter(config.mangle?.include, config.mangle?.exclude)
    expect(filter('xx/yy.js?a=1')).toBe(true)
  })
})
