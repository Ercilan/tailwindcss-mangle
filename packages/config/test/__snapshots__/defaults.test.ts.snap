// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`defaults > getDefaultPatchConfig 1`] = `
{
  "output": {
    "filename": ".tw-patch/tw-class-list.json",
    "loose": true,
    "removeUniversalSelector": true,
  },
  "tailwindcss": {},
}
`;

exports[`defaults > getDefaultUserConfig 1`] = `
{
  "mangle": {
    "classListPath": ".tw-patch/tw-class-list.json",
    "classMapOutput": {
      "enable": false,
      "filename": ".tw-patch/tw-map-list.json",
      "loose": true,
    },
    "disabled": false,
    "exclude": [],
    "include": [
      /\\\\\\.\\(html\\|js\\|ts\\|jsx\\|tsx\\|vue\\|svelte\\|astro\\|elm\\|php\\|phtml\\|mdx\\|md\\)\\(\\?:\\$\\|\\\\\\?\\)/,
      /\\\\\\.\\(css\\|less\\|sass\\|scss\\|styl\\|stylus\\|pcss\\|postcss\\|sss\\)\\(\\?:\\$\\|\\\\\\?\\)/,
    ],
    "mangleClassFilter": [Function],
    "preserveFunction": [],
  },
  "patch": {
    "output": {
      "filename": ".tw-patch/tw-class-list.json",
      "loose": true,
      "removeUniversalSelector": true,
    },
    "tailwindcss": {},
  },
}
`;
